app/queries/find_exact_match.rb

```diff
@@ -6,37 +6,14 @@ def initialize(query_field: :tags, scope: SubscriberList)
     @scope = scope
   end
 
-  def call(query_hash)
-    return [] unless query_hash.present?
+  def call(tags_links)
+    return [] unless tags_links.present?
 
-    subscriber_lists = subscriber_lists_where_all_keys_present(query_hash)
-
-    subscriber_lists.select do |subscriber_list|
-      subscriber_list_tags_or_links = subscriber_list.send(@query_field) # send ensures the keys are symbols
-
-      subscriber_list_tags_or_links.all? do |key, subscriber_list_tag_or_link|
-        query_any = query_hash[key].fetch(:any, [])
-        query_all = query_hash[key].fetch(:all, [])
-        links_or_tags_any = subscriber_list_tag_or_link.fetch(:any, [])
-        links_or_tags_all = subscriber_list_tag_or_link.fetch(:all, [])
-        query_any.sort == links_or_tags_any.sort && query_all.sort == links_or_tags_all.sort
-      end
-    end
+    digest = HashDigest.generate(tags_links, query_field)
+    scope.where(tags_links_digest: digest).first
```

Comment:
Given there is a risk of collisions (albeit minute) should this pull out all results that match digest and then do a final sanity check on which is equal? or is that overkill?